gc/base/MemoryPoolAddressOrderedList.cpp

```diff
@@ -634,9 +669,29 @@ MM_MemoryPoolAddressOrderedList::internalAllocateTLH(MM_EnvironmentBase *env, ui
 	consumedSize = (maximumSizeInBytesRequired > freeEntrySize) ? freeEntrySize : maximumSizeInBytesRequired;
 	_largeObjectAllocateStats->decrementFreeEntrySizeClassStats(freeEntrySize);
 
-	/* If the leftover chunk is smaller than the minimum size, hand it out */
+	if (!alignTLHForGC(env, freeEntry, &consumedSize)) {
```

Comment:
I'm wondering if we could do this slightly before with motivation of having only one common spot where we jump to check if needed and actually perform the alignment?